jstz_cli/src/account/mod.rs

```diff
@@ -72,17 +87,24 @@ fn delete_account(alias: String, cfg: &mut Config) -> Result<()> {
 }
 
 pub fn login(alias: String, cfg: &mut Config) -> Result<()> {
-    let account = cfg.accounts().get(&alias)?;
+    if !cfg.accounts().contains(&alias) {
+        return Err(anyhow!("Account not found"));
+    }
+    if cfg.accounts().current_alias == Some(alias.clone()) {
+        return Err(anyhow!("Already logged in to account {}!", alias));
+    }
 
-    println!(
-        "Logged in to account {} with address {}",
-        account.alias, account.address
-    );
+    let account = cfg.accounts().get(&alias)?;
+    if let Account::Owned(OwnedAccount { alias, address, .. }) = account {
+        println!("Logged in to account {} with address {}", alias, address);
 
-    cfg.accounts().current_alias = Some(alias);
-    cfg.save()?;
+        cfg.accounts().current_alias = Some(alias.to_string());
+        cfg.save()?;
 
-    Ok(())
+        Ok(())
+    } else {
+        return Err(anyhow!("Account is not owned"));
+    }
```

Comment:
I expected an `into_owned_account` and `into_alias_account` methods on `Account` to avoid this boilerplate.